spyder_line_profiler/spyder/plugin.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# Copyright © 2021, Spyder Line Profiler 5
+#
+# Licensed under the terms of the MIT license
+# ----------------------------------------------------------------------------
+"""
+Spyder Line Profiler 5 Plugin.
+"""
+
+# Third-party imports
+from qtpy.QtGui import QIcon
+
+# Spyder imports
+from spyder.api.plugins import Plugins, SpyderDockablePlugin
+from spyder.api.translations import get_translation
+from spyder.api.plugin_registration.decorators import on_plugin_available
+
+# Local imports
+from spyder_line_profiler.spyder.confpage import SpyderLineProfilerConfigPage
+from spyder_line_profiler.spyder.widgets import SpyderLineProfilerWidget
+
+_ = get_translation("spyder_line_profiler.spyder")
+
+
+class SpyderLineProfiler(SpyderDockablePlugin):
+    """
+    Spyder Line Profiler plugin for Spyder 5.
+    """
+
+    NAME = "spyder_line_profiler"
+    REQUIRES = [Plugins.Editor]
+    OPTIONAL = []
+    TABIFY = Plugins.Help
+    WIDGET_CLASS = SpyderLineProfilerWidget
+    CONF_SECTION = NAME
+    CONF_WIDGET_CLASS = SpyderLineProfilerConfigPage
+
+    # --- Signals
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    @staticmethod
+    def get_name():
+        return _("Line Profiler")
+
+    def get_description(self):
+        return _("Line profiler display for Spyder 5")
+
+    def get_icon(self):
+        return QIcon('./data/images/spyder.line_profiler.png')
+
+
+    def on_initialize(self):
+        widget = self.get_widget()
+
+        # run_action = self.create_action(
+        #     ProfilerActions.ProfileCurrentFile,
+        #     text=_("Run profiler"),
+        #     tip=_("Run profiler"),
+        #     icon=self.create_icon('profiler'),
+        #     triggered=self.run_profiler,
+        #     register_shortcut=True,
+        # )
+
+        # run_action.setEnabled(is_profiler_installed())
```

Comment:
To add the action to the corresponding menu I think you could do something like:

```python
self._main.run_menu_actions += [lineprofiler_act]
```
In the new API the plugins have a reference to the main window through `self._main` instead of `self.main`

Reply:
I don't think we need this, or what is the purpose of this?

Reply:
I think for the moment preserving the way the plugin was working is better and also seems that some people use it so I think is worthy to keep it as for the issues here: https://github.com/spyder-ide/spyder-line-profiler/issues/57

However, maybe later we can check the removal of the action from the menu.

To create the entry in the menu then we can remove the lines here and instead add a method to handle the connection with the main menu as is done in the Profiler plugin: https://github.com/spyder-ide/spyder/blob/e948501d201d77c1dff092bdc784310211f452d0/spyder/plugins/pylint/plugin.py#L126-L132